plugin/storage/es/samplingstore/storage_test.go

```diff
@@ -0,0 +1,417 @@
+			writeService := &mocks.IndexService{}
+			w.client.On("Index").Return(writeService)
+			w.client.On("GetVersion").Return(testCase.esVersion)
+
+			writeService.On("Index", stringMatcher(indexName)).Return(writeService)
+			writeService.On("Type", stringMatcher(throughputType)).Return(writeService)
+			writeService.On("BodyJson", mock.Anything).Return(writeService)
+			writeService.On("Add", mock.Anything).Return(nil, testCase.writeError)
+			err := w.storage.InsertThroughput(throughputs)
+			if testCase.expectedError != "" {
+				require.EqualError(t, err, testCase.expectedError)
+			} else {
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
+func stringMatcher(q string) interface{} {
+	matchFunc := func(s string) bool {
+		return strings.Contains(s, q)
+	}
+	return mock.MatchedBy(matchFunc)
+}
+
+func TestInsertProbabilitiesAndQPS(t *testing.T) {
+	testCases := []struct {
+		writeError    error
+		expectedError string
+		esVersion     uint
+	}{
+		{
+			expectedError: "",
+			esVersion:     6,
+		},
+		{
+			expectedError: "",
+			esVersion:     7,
+		},
+	}
+	for _, testCase := range testCases {
+		withEsSampling("", "2006-01-02", defaultMaxDocCount, func(w *samplingStorageTest) {
+			pAQ := dbmodel.ProbabilitiesAndQPS{
+				Hostname:      "dell11eg843d",
+				Probabilities: samplemodel.ServiceOperationProbabilities{"new-srv": {"op": 0.1}},
+				QPS:           samplemodel.ServiceOperationQPS{"new-srv": {"op": 4}},
+			}
+			fixedTime := time.Now()
+			indexName := indexWithDate("", "2006-01-02", fixedTime)
+			writeService := &mocks.IndexService{}
+			w.client.On("Index").Return(writeService)
+			w.client.On("GetVersion").Return(testCase.esVersion)
+
+			writeService.On("Index", stringMatcher(indexName)).Return(writeService)
+			writeService.On("Type", stringMatcher(probabilitiesType)).Return(writeService)
+			writeService.On("BodyJson", mock.Anything).Return(writeService)
+			writeService.On("Add", mock.Anything).Return(nil, testCase.writeError)
+			err := w.storage.InsertProbabilitiesAndQPS(pAQ.Hostname, pAQ.Probabilities, pAQ.QPS)
+			if testCase.expectedError != "" {
+				require.EqualError(t, err, testCase.expectedError)
+			} else {
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestGetThroughput(t *testing.T) {
+	mockIndex := "jaeger-sampling-" + time.Now().UTC().Format("2006-01-02")
+	goodThroughputs := `{
+		"timestamp": "2024-02-08T12:00:00Z",
+		"throughputs": [
+			{
+				"Service": "my-svc",
+				"Operation": "op",
+				"Count": 10
+			},
+			{
+				"Service": "another-svc",
+				"Operation": "another-op",
+				"Count": 20
+			}
+		]
+	}`
+	badThroughputs := `badJson{hello}world`
+	testCases := []struct {
+		searchResult   *elastic.SearchResult
+		searchError    error
+		expectedError  string
+		expectedOutput []*samplemodel.Throughput
+		indexPrefix    string
+		maxDocCount    int
+		indices        []interface{}
+	}{
+		{
+			searchResult: createSearchResult(goodThroughputs),
```

Comment:
it's customary to have a `name` property for the tests that makes it easier to locate in case of failure. Right now your subtests will be named /1, /2, ...